docs/datacite.README.md

```diff
@@ -0,0 +1,121 @@
+# DataCite DOI implementation
+
+## DataCite Documentation and References
+
+- DataCite Rest API Guide: https://support.datacite.org/docs/api
+- DataCite REST API Reference: https://support.datacite.org/reference/introduction
+- DataCite Metadata Schema 4.4: https://schema.datacite.org/meta/kernel-4.4/
+
+## DataCite background info
+
+### DOI states, EZID vs DataCite
+
+- reserved (EZID) <==> draft (DataCite)
+- public (EZID) <==> findable (DataCite)
+- unavailable (EZID) <==> registered (DataCite)
+
+### Required DataCite Properties
+
+DataCite requires the following properties for a DOI in the findable state (including minting a findable DOI)
+
+- creators
+- publisher
+- publication year
+- resource type general (controlled?)
+- title
+- url
+
+## Introduction to the overall implementation of DataCite DOIs in Hyacinth
+
+There are three main classes in this implementation
+
+- Datacite::RestApi::V2::Api
+This class is the direct interface to the DataCite REST API. It includes three methods, one each for the GET, POST, and PUT requests that can
+be sent to the DataCite REST API. If included/relevant, the request body is given as an argument in JSON format.
+- Datacite::RestApi::V2::Data
+This class is used to populate and construct the JSON structure that will be sent via a request body to the DataCite REST API (see above)
+- Datacite
+This is the class that is used by the rest of Hyacinth to mint and update DOIs. It makes use of the above two classes. This class also contains
+the mapping functionality between Hyacinth fields and DataCite properties.
+
+There is also a legacy class, Datacite::Metadata
+This class wraps the Digital Object Data from a DigitalObject and provides methods to access metadata.
+NOTE: this class SHOULD move out of the datacite subdir since it is independent of the datacite functionality. Also, I would prefer to
```

Comment:
If easy, can this be done now as part of this PR?

Reply:
`ExternalIdentifierAdapter::Datacite::Metadata` changed to `ExternalIdentifierAdapter::HyacinthMetadata`